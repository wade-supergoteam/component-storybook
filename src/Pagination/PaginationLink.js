// @flow

import styled, { css } from "styled-components";

import themeProp from "../utils/theme";

import { hoverFocus } from "../utils/hover";

import {
  paginationPaddingY,
  paginationPaddingX,
  paginationLineHeight,
  paginationColor,
  paginationBg,
  paginationBorderColor,
  paginationHoverColor,
  paginationHoverBg,
  paginationHoverBorderColor,
  paginationBorderWidth,
} from "./default-theme";

const PaginationLink = styled.a`
  position: relative;
  display: block;
  padding: ${themeProp("paginationPaddingY", paginationPaddingY)} ${themeProp("paginationPaddingX", paginationPaddingX)};
  margin-left: -1px;
  line-height: ${themeProp("paginationLineHeight", paginationLineHeight)};
  color: ${themeProp("paginationColor", paginationColor)};
  background-color: ${themeProp("paginationBg", paginationBg)};
  border: ${themeProp("paginationBorderWidth", paginationBorderWidth)} solid
    ${themeProp("paginationBorderColor", paginationBorderColor)};

  ${hoverFocus(css`
    color: ${themeProp("paginationHoverColor", paginationHoverColor)};
    text-decoration: none;
    background-color: ${themeProp("paginationHoverBg", paginationHoverBg)};
    border-color: ${themeProp("paginationHoverBorderColor", paginationHoverBorderColor)};
  `)};
`;

export default PaginationLink;
