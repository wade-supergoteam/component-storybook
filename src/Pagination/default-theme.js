// @flow

// general variables
export const fontSizeLg = "1.25rem";
export const lineHeightLg = "1.5";
export const borderRadiusLg = ".3rem";
export const fontSizeSm = ".875rem";
export const lineHeightSm = "1.5";
export const borderRadiusSm = ".2rem";

// specific pagination variables
export const paginationPaddingY = ".5rem";
export const paginationPaddingX = ".75rem";
export const paginationPaddingYsm = ".25rem";
export const paginationPaddingXsm = ".5rem";
export const paginationPaddingYlg = ".75rem";
export const paginationPaddingXlg = "1.5rem";
export const paginationLineHeight = "1.25";
export const paginationColor = "#007bff";
export const paginationBg = "#fff";
export const paginationBorderWidth = "1px";
export const paginationBorderColor = "#dee2e6";
export const paginationFocusBoxShadow = "0 0 0 .2rem rgba(0,123,255,0.25)";
export const paginationHoverColor = "#0056b3";
export const paginationHoverBg = "#e9ecef";
export const paginationHoverBorderColor = "#dee2e6";
export const paginationActiveColor = "#fff";
export const paginationActiveBg = "#007bff";
export const paginationActiveBorderColor = "#007bff";
export const paginationDisabledColor = "#6c757d";
export const paginationDisabledBg = "#fff";
export const paginationDisabledBorderColor = "#dee2e6";
