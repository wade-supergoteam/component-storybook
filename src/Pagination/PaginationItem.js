// @flow

import styled, { css } from "styled-components";

import themeProp from "../utils/theme";

import { borderRightRadius, borderLeftRadius } from "../utils/border-radius";

import PaginationLink from "./PaginationLink";

import {
  paginationActiveColor,
  paginationActiveBg,
  paginationActiveBorderColor,
  paginationDisabledColor,
  paginationDisabledBg,
  paginationDisabledBorderColor,
} from "./default-theme";

const PaginationItem = styled.li`
  &:first-child {
    ${PaginationLink} {
      margin-left: 0;
      ${borderLeftRadius("0.25rem")};
    }
  }

  &:last-child {
    ${PaginationLink} {
      ${borderRightRadius("0.25rem")};
    }
  }

  ${props =>
    props.active &&
    css`
      ${PaginationLink} {
        z-index: 2;
        color: ${themeProp("paginationActiveColor", paginationActiveColor)};
        background-color: ${themeProp("paginationActiveBg", paginationActiveBg)};
        border-color: ${themeProp("paginationActiveBorderColor", paginationActiveBorderColor)};
      }
    `};

  ${props =>
    props.disabled &&
    css`
      ${PaginationLink} {
        color: ${themeProp("paginationDisabledColor", paginationDisabledColor)};
        pointer-events: none;
        background-color: ${themeProp("paginationDisabledBg", paginationDisabledBg)};
        border-color: ${themeProp("paginationDisabledBorderColor", paginationDisabledBorderColor)};
      }
    `};
`;

export default PaginationItem;
