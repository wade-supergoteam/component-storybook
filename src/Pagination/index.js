import PaginationItem from "./PaginationItem";
import PaginationLink from "./PaginationLink";

export { PaginationItem, PaginationLink };
export { default } from "./Pagination";
