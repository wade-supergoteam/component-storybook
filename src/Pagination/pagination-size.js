// @flow

import { css } from "styled-components";

import { borderLeftRadius, borderRightRadius } from "../utils/border-radius";

import PaginationLink from "./PaginationLink";
import PaginationItem from "./PaginationItem";

export function paginationSize(paddingY: string, paddingX: string, fontSize: string, lineHeight: string, borderRadius: string) {
  return css`
    ${PaginationLink} {
      padding: ${paddingY} ${paddingX};
      font-size: ${fontSize};
    }

    ${PaginationItem} {
      &:first-child {
        ${PaginationLink} {
          ${borderLeftRadius(borderRadius)};
        }
      }
      &:last-child {
        ${PaginationLink} {
          ${borderRightRadius(borderRadius)};
        }
      }
    }
  `;
}

export default paginationSize;
