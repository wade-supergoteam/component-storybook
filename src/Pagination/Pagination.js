// @flow

import styled from "styled-components";

// $FlowIssue
import get from "lodash.get";

import { borderRadius } from "../utils/border-radius";

import { paginationSize } from "./pagination-size";
import listUnstyled from "../utils/lists";

import PaginationLink from "./PaginationLink";
import PaginationItem from "./PaginationItem";

import {
  paginationPaddingYsm,
  paginationPaddingXsm,
  paginationPaddingYlg,
  paginationPaddingXlg,
  fontSizeLg,
  lineHeightLg,
  borderRadiusLg,
  fontSizeSm,
  lineHeightSm,
  borderRadiusSm,
} from "./default-theme";

const Pagination = styled.ul`
  display: flex;
  ${listUnstyled};
  ${borderRadius()};

  ${({ size, theme }) =>
    size === "large" &&
    paginationSize(
      get(theme, "paginationPaddingYlg", paginationPaddingYlg),
      get(theme, "paginationPaddingXlg", paginationPaddingXlg),
      get(theme, "fontSizeLg", fontSizeLg),
      get(theme, "lineHeightLg", lineHeightLg),
      get(theme, "borderRadiusLg", borderRadiusLg),
    )};

  ${({ size, theme }) =>
    size === "small" &&
    paginationSize(
      get(theme, "paginationPaddingYsm", paginationPaddingYsm),
      get(theme, "paginationPaddingXsm", paginationPaddingXsm),
      get(theme, "fontSizeSm", fontSizeSm),
      get(theme, "lineHeightSm", lineHeightSm),
      get(theme, "borderRadiusSm", borderRadiusSm),
    )};
`;

Pagination.Link = PaginationLink;
Pagination.Item = PaginationItem;

Pagination.defaultProps = {};

Pagination.propTypes = {};

export default Pagination;
