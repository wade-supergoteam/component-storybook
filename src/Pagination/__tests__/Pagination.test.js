import React from "react";
import renderer from "react-test-renderer";
import PaginationItem from "../PaginationItem";
import Pagination from "../";

test("Default Pagination", () => {
  const wrapper = renderer.create(<Pagination />);
  expect(wrapper).toMatchSnapshot();
});

test("Pagination large", () => {
  const wrapper = renderer.create(<Pagination size="large" />);
  expect(wrapper).toMatchSnapshot();
});

test("Pagination large", () => {
  const wrapper = renderer.create(<Pagination size="small" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create Pagination Item active", () => {
  const wrapper = renderer.create(<PaginationItem active />);
  expect(wrapper).toMatchSnapshot();
});

test("Create Pagination Item disabled", () => {
  const wrapper = renderer.create(<PaginationItem disabled />);
  expect(wrapper).toMatchSnapshot();
});
