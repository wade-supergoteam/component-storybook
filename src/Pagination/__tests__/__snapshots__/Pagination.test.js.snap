// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Pagination Item active 1`] = `
<li
  className="sc-bwzfXH kVzdrT"
/>
`;

exports[`Create Pagination Item disabled 1`] = `
<li
  className="sc-bwzfXH gEzWvi"
  disabled={true}
/>
`;

exports[`Default Pagination 1`] = `
<ul
  className="sc-htpNat eyTibV"
/>
`;

exports[`Pagination large 1`] = `
<ul
  className="sc-htpNat bQjkvL"
  size="large"
/>
`;

exports[`Pagination large 2`] = `
<ul
  className="sc-htpNat fjLBcq"
  size="small"
/>
`;
